### DISCUSSION TITLE
Minimise app

### DISCUSSION THREAD

**Discussion by raresn** [airsdk/Adobe-Runtime-Support#2784] [General]
Any way to minimise an app on all platforms, @ajwfrost ?
I've been looking for quite some time for a solution and i came up with nothing.
Maybe could be a feature in the Application ANE, @marchbold - though would be cool to have it directly from AIR.

**Comment by ajwfrost** [comment]
Hi

So on desktops I assume this is already fine, but you're looking for a way to push an application into the background when it's running on a mobile?

I checked with the team and someone just mentioned this may be considered as 'bad behaviour' by Apple - a while ago, he'd had an app rejected from the App Store because of this ..
https://stackoverflow.com/a/5362887/11433244

So, not 100% sure about this, might be worth trying it out with a quick ANE before we looked at incorporating this capability into the runtime!

thanks

**Reply by raresn** [reply]
@ajwfrost Yes, the main thing here was on android, where the user presses the "android back button" a few times, we display on screen "press back 2 more times to minimise" - and when the user does, we kill the app instead of minimising it. (send it to background)
So really a matter of Android if i think about it better.

**Reply by itlancer** [reply]
@raresn 
May be in your case you can handle `KeyboardEvent.KEY_DOWN` events, then check `KeyboardEvent::keyCode` for `Keyboard.BACK` and call `KeyboardEvent::preventDefault()` in event listener first few times to "block" application minimization (which default by OS)? And with these few times show "press back 2 more times to minimise".

**Reply by raresn** [reply]
@itlancer the wanted behaviour is to minimise the app when the user presses back 3 times.

**Reply by marchbold** [reply]
Hi @raresn , this is definitely something you can achieve within AIR on Android and using the keyboard events and the  `KeyboardEvent::preventDefault()` as @itlancer suggests. The `preventDefault` method will stop the back event from minimising your app so you'd setup some conditions under which this is called and then let it flow normally when you want your app to minimise. 

For example: 

```as3
var count:int = 0;

function keyDownHandler( event:KeyboardEvent ):void
{
	if (event.keyCode == Keyboard.BACK)
	{
		count ++;
		if (count < 3)
			event.preventDefault();
	}
}

NativeApplication.nativeApplication.addEventListener( KeyboardEvent.KEY_DOWN, keyDownHandler );

```

You'll need to add some logic as to when to reset the count etc (eg a timer on the first back call). 

Regarding a "force minimise" call I agree with @ajwfrost here in that you will likely face review issues as you are potentially providing a non-standard navigation and both Google and Apple have issues with this. Even with the above code I would suggest caution on Android unless you make it clear to users that they need to navigate back multiple times.
